src/nsls2api/services/proposal_service.py

```diff
@@ -336,18 +426,12 @@ async def has_valid_cycle(proposal: Proposal):
     # proposal then the cycle information is invalid
     return not (
         (len(proposal.cycles) == 0)
-        and (
-            proposal.pass_type_id != "300005"
-            or proposal.pass_type_id != "300042"
-        )
+        and (proposal.pass_type_id != "300005" or proposal.pass_type_id != "300042")
     )
```

Comment:
I think the reverse logic is simpler here, and can reuse `is_commissioning()`.
`not(A and B) == (not A) or (not B)`

```suggestion
    return (
         (len(proposal.cycles) > 0)
         or (await is_commissioning(proposal))
     )
```